### ISSUE TITLE
Incorrect URL encoding

### ISSUE THREAD

**Issue by zipme** [Mikhus/domurl#29]
When init with url like this `new Url('http://www.example.com/hello world here.jpg').toString()`

returns: `http://www.example.com/hello%2520world%2520here.jpg`

expected result: 'http://www.example.com/hello%20world%20here.jpg'

Not sure why the space got encoded into `%2520`

When remove the protocol it works correctly:

`new Url('www.example.com/hello world here.jpg').toString()`
=> `'www.example.com/hello%20world%20here.jpg'`

**Comment by sindilevich** [comment]
@zipme, it looks like the URI got encoded twice. The first time it became: `... hello%20world%20here ...`. And then the second pass has encoded the `%` character and it became: `... hello%2520world%2520here ...`. `%25` is the encoded version of the `%`character.

**Comment by zipme** [comment]
@sindilevich indeed, do you know if there is a quick fix for this?

**Comment by sindilevich** [comment]
@zipme, do you use this library with node.js?

**Comment by zipme** [comment]
@sindilevich I use it in browser

**Comment by sindilevich** [comment]
@zipme, I built an HTML page as follows, and used the latest url.js file as of today. It worked as expected for me: `... hello%20world%20here ...`.

The HTML page:

```html
<html>

<head>
    <title>DOM URI test page</title>
    <script src="url.js"></script>
</head>

<body>
    <label>
        Enter an URI to parse:
        <input type="url" value="http://www.example.com/hello world here.jpg" class="uri-input" style="width: 100%;" />
    </label>
    <br />
    <br />
    <br />
    <label>
        The parsed URI is:
        <textarea class="uri-output" cols="80" rows="2" style="width: 100%;"></textarea>
    </label>
    <br />
    <br />
    <br />
    <button class="act-button">Act on URI</button>
    <script>
        const actButton = document.querySelector(".act-button");

        actButton.addEventListener("click", () => { 
            const uriInput = document.querySelector(".uri-input");
            const uriOutput = document.querySelector(".uri-output");
            const uri = uriInput.value;

            if (uri !== "") {
                uriOutput.value = (new URL(uri)).toString();
            }
        }, false);
    </script>
</body>

</html>
```

**Comment by zipme** [comment]
@sindilevich Indeed, I tried this in fiddler it does work correctly, I am using the library in a react project and here is the code snippet:

`import Url from 'domurl'`
`new Url('http://www.example.com/hello world here.jpg').toString()`
 => 'http://www.example.com/hello%2520world%2520here.jpg'

**Comment by step307** [comment]
I've debugged a bit and found the function Url.prototype.paths a bit strange.

It seems to work as a setter for this.path by given array or as getter returning an array if nothing passed.

It looks like this function expects unescaped values.

More over the only call of this function seems to me useless:
`self.paths((self.path.charAt(0) === '/' ?
            self.path.slice(1) : self.path).split('/')
        );`

Basically we set this.path by array parsed from this.path. And in this case the function gets already escaped values which is not correct. 

May be it should be something like:
`self.paths(self.paths());`
Which makes even less sense on my opinion

Whether this is required on some browsers/engines for some magic or it can be skipped.

If I delete this function or its call, all tests are passed in chrome, ff and ie 11

Does anyone have any idea ?

**Comment by step307** [comment]
Ok, not its clear IE needs this function to add starting "/"

**Comment by step307** [comment]
Pull request: #32
